systest/Makefile

```diff
@@ -1,119 +1,178 @@
-singlebigip_results := $(RESULTSDIR)/$(singlebigip_session)
-
-
-testenv_config := bigip.testenv.yaml
-
-VENV := buildbot
-VENV_ACTIVATE := $(MAKEFILE_DIR)/$(VENV)/bin/activate
-
-# Before we run any tests we need a virtualenv and a few packages
-pre-build:
-	sudo -E apt-get update; \
-	sudo -E apt-get install -y libssl-dev; \
-	sudo -E apt-get install -y libffi-dev; \
-	sudo -E -H pip install --upgrade pip; \
-	sudo -E -H pip install tox; \
-	sudo -E -H pip install git+ssh://git@bldr-git.int.lineratesystems.com/tools/testenv.git; \
-	sudo -E -H pip install git+ssh://git@bldr-git.int.lineratesystems.com/velcro/systest-common.git; \
-	sudo -E -H pip install git+ssh://git@bldr-git.int.lineratesystems.com/tools/pytest-meta.git; \
-	sudo -E -H pip install git+ssh://git@bldr-git.int.lineratesystems.com/tools/pytest-autolog.git; \
-	sudo -E -H pip install git+ssh://git@bldr-git.int.lineratesystems.com/tools/pytest-symbols.git;
-
-unit: pre-build
-	cd $(MAKEFILE_DIR)/../; \
+TAGINFO := $(shell git describe --long)
+
+# git branch
+BRANCH := $(shell git rev-parse --abbrev-ref HEAD)
+
+# /ASB/PATH/systest/test_results: This is where the buildbot system will look
+# for test reuslts.
+RESULTS_DIR := $(MAKEFILE_DIR)/test_results
+
+
+#### INITIALIZATION SECTION
+#    Since all tests will run on the same buildbot worker, and we do not (yet)
+#    require different packages for different suites, all packages are
+#    installed.  Each deploy maps to its own GUMBALLS project.  This section
+#    has one because unittests are run in this section.  
+#    Tests use tox for python variants.
+functest: 
+	-@echo executing $@
+	sudo -E -H $(SCRIPT_DIR)/install_test_infra.sh &&
+	export GUMBALLS_PROJECT=$(PROJNAME)_$(BRANCH)-unit &&
+	export RESULTS_DIR=$(RESULTS_DIR)/$${GUMBALLS_PROJECT} &&
 	tox -e unit --sitepackages -- \
 		--exclude incomplete no_regression \
-		--autolog-outputdir $(RESULTSDIR) \
-		--autolog-session $(unit_session) \
-
-functest:
-	$(MAKE) -C . unit
-	$(MAKE) -j -C . functest_all
-
-functest_all:
-	@echo "automated functional tests..."
-	$(MAKE) -C . disconnected_service
+		--autolog-outputdir $${RESULTS_DIR} \
+		--autolog-session unit_$(TAGINFO)_`date +"%Y%m%d-%H%M%S"` \
+	    $(PROJDIR)/f5_openstack_agent &&
+	$(MAKE) -C . $(DEPLOYS)
+
+#### MULTIPLEXING SECTION:
+#    there are (DEVICES * CLOUDTYPES) rules in this section. They have targets
+#    like '12_1_1_overcloud' and '11_6_1_undercloud'.
+
+#  This rule sets variables to values specific for a particular deploy. This
+#  is where interfacing software (trtl) names are set.
+#  "GUMBALLS_PROJECT" maps to a raw in gumballs (URL)XXX
+#  "GUMBALLS_SESSION" maps to a column in gumballs" (URL)XXX
+$(DEPLOYS):
+	-@echo executing $@
+	. $(MAKEFILE_DIR)/device_version_maps.sh &&
+	export DEVICE=BIGIP_`python -c 'print("$@".split("-")[0])'` &&
+	export CLOUD=`python -c 'print("$@".split("-")[1])'` &&
+	export GUMBALLS_PROJECT=$(PROJNAME)_$(BRANCH)-$@ &&
```

Comment:
Looks like the gumballs project is being named the same as the automatic variable for the target name. That target name is using hyphens instead of underscores. Since these names are important, any reason why we are deviating from what driver is doing?